packages/mui-base/src/Portal/Portal.tsx

```diff
@@ -8,19 +8,29 @@ import {
   unstable_useForkRef as useForkRef,
   unstable_setRef as setRef,
 } from '@mui/utils';
+import { PortalProps } from './Portal.types';
 
-function getContainer(container) {
+function getContainer(container: Element | (() => Element | null) | null) {
   return typeof container === 'function' ? container() : container;
 }
 
 /**
  * Portals provide a first-class way to render children into a DOM node
  * that exists outside the DOM hierarchy of the parent component.
+ *
+ * Demos:
+ *
+ * - [Portal](https://mui.com/base/react-portal/)
+ *
+ * API:
+ *
+ * - [Portal API](https://mui.com/base/api/portal/)
  */
-const Portal = React.forwardRef(function Portal(props, ref) {
-  const { children, container, disablePortal = false } = props;
-  const [mountNode, setMountNode] = React.useState(null);
-  const handleRef = useForkRef(React.isValidElement(children) ? children.ref : null, ref);
+
+const Portal = React.forwardRef(function Portal(props: PortalProps, ref:React.ForwardedRef<Element>) {
+  const { children, container = null, disablePortal = false } = props;
+  const [mountNode, setMountNode] = React.useState<Element | (() => Element | null) | null>(null);
```

Comment:
Why do we say that `mountNode` is `Element | (() => Element | null) | null`
If after that we do `as HTMLElement` ?

The two should be coherent.

Unless I'm missing something, the `container` prop could be `HTMLElement | () => HTMLElement | null) | null` (if we refer to its description).
And we never set a function in `mountNode` since we resolve the function version of `container` in `getContainer`.

So for me `mountNode` is of type `HTMLElement | null`.

Reply:
That makes sense I will change that.